examples/gym_ex/rl/agent.py

```diff
@@ -172,13 +172,19 @@ def fit(self, env: gym.Env, nb_steps: int) -> None:
         :return: None
         """
         action_counter = 0
+        episode_counter = 0
+        nb_steps_digits = len(str(nb_steps))
 
-        env.reset()
         while action_counter < nb_steps:
```

Comment:
Why do we need this outer while loop?; also, I am not sure what the episode counter does... How can it ever be more than 1?

Reply:
The `BanditEnv` is a very special case in that sense since there is no concept of "episode": there's no need to reset the environment at the initial conditions.
However, in the very vast majority of RL environments, there is the need to reset the environment and start the simulation again. Think about Atari games: whenever the agent loses, in order to continue the learning, a new episode must start.